src/Widgets/AppContextMenu.vala

```diff
@@ -108,6 +109,24 @@ public class Slingshot.AppContextMenu : Gtk.Menu {
         });
     }
 
+    private void open_in_appcenter () {
+        try {
+            Process.spawn_command_line_async ("xdg-open appstream://" + appstream_comp_id);
```

Comment:
Instead of a commandline, probably better to use [GLib.AppInfo.launch_default_for_uri](https://valadoc.org/gio-2.0/GLib.AppInfo.launch_default_for_uri.html)

Reply:
@danrabbit I have made the change however I am wondering If I should use launch_default_for_uri_async and catch the error as earlier and show a dialog?